src/dashboard/src/components/administration/forms.py

```diff
@@ -129,10 +139,306 @@ class AtomDipUploadSettingsForm(SettingsForm):
+        'required': False,
+        'initial': None,
+    }
+
+    name = forms.RegexField(max_length=16, regex=r'^\w+$', required=True)
+
+    def __init__(self, *args, **kwargs):
+        super(ProcessingConfigurationForm, self).__init__(*args, **kwargs)
+        for choice_uuid, field in self.processing_fields.iteritems():
+            ftype = field['type']
+            opts = self.DEFAULT_FIELD_OPTS.copy()
+            if 'label' in field:
+                opts['label'] = field['label']
+            if ftype == 'days':
+                if 'min_value' in field:
+                    opts['min_value'] = field['min_value']
+                self.fields[choice_uuid] = forms.IntegerField(**opts)
+                if 'placeholder' in field:
+                    self.fields[choice_uuid].widget.attrs['placeholder'] = field['placeholder']
+            else:
+                choices = opts['choices'] = list(self.EMPTY_CHOICES)
+                if ftype == 'boolean':
+                    if 'yes_option' in field:
+                        choices.append((field['yes_option'], 'Yes'))
+                    if 'no_option' in field:
+                        choices.append((field['no_option'], 'No'))
+                elif ftype == 'chain_choice':
+                    link = models.MicroServiceChainLink.objects.get(pk=choice_uuid)
+                    chain_choices = models.MicroServiceChainChoice.objects.filter(choiceavailableatlink_id=link.pk)
+                    ignored_choices = field.get('ignored_choices')
+                    for item in chain_choices:
+                        chain = item.chainavailable
+                        if ignored_choices is not None:
+                            if chain.description in ignored_choices:
+                                continue
+                        choices.append((chain.pk, chain.description))
+                elif ftype == 'replace_dict':
+                    link = models.MicroServiceChainLink.objects.get(pk=choice_uuid)
+                    replace_dicts = models.MicroServiceChoiceReplacementDic.objects.filter(choiceavailableatlink_id=link.pk)
+                    for item in replace_dicts:
+                        choices.append((item.pk, item.description))
+                elif ftype == 'storage_service':
+                    for loc in get_storage_locations(purpose=field['purpose']):
+                        choices.append((loc['resource_uri'], loc['description']))
+                self.fields[choice_uuid] = forms.ChoiceField(**opts)
+
+    def load_config(self, name):
+        """
+        Bound the choices found in the XML document to the form fields.
+        """
+        self.fields['name'].initial = name
+        self.fields['name'].widget.attrs['readonly'] = 'readonly'
+        config_path = os.path.join(helpers.processing_config_path(), '{}ProcessingMCP.xml'.format(name))
+        with open(config_path, 'r') as f:
+            xml = f.read()
+            root = etree.fromstring(xml)
```

Comment:
Tip: Can combine these 2 lines into `root = etree.parse(f)`  The rest of the block doesn't have to be within the `with` block, since the file's already been read from.